demo_nodes_cpp/CMakeLists.txt

```diff
@@ -77,51 +77,149 @@ add_dependencies(services_library)
 add_dependencies(parameters_library)
 add_dependencies(topics_library)
 
-rclcpp_components_register_node(timers_library
-  PLUGIN "demo_nodes_cpp::OneOffTimerNode"
-  EXECUTABLE one_off_timer)
-rclcpp_components_register_node(timers_library
-  PLUGIN "demo_nodes_cpp::ReuseTimerNode"
-  EXECUTABLE reuse_timer)
-
-rclcpp_components_register_node(services_library
-  PLUGIN "demo_nodes_cpp::ServerNode"
-  EXECUTABLE add_two_ints_server)
-rclcpp_components_register_node(services_library
-  PLUGIN "demo_nodes_cpp::ClientNode"
-  EXECUTABLE add_two_ints_client_async)
-
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ListParameters"
-  EXECUTABLE list_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterBlackboard"
-  EXECUTABLE parameter_blackboard)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::SetAndGetParameters"
-  EXECUTABLE set_and_get_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterEventsAsyncNode"
-  EXECUTABLE parameter_events_async)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::EvenParameterNode"
-  EXECUTABLE even_parameters_node)
-
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Talker"
-  EXECUTABLE talker)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageTalker"
-  EXECUTABLE talker_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Listener"
-  EXECUTABLE listener)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageListener"
-  EXECUTABLE listener_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::ListenerBestEffort"
-  EXECUTABLE listener_best_effort)
+rclcpp_components_register_nodes(
+  PLUGINS
+    "demo_nodes_cpp::ReuseTimerNode"
+    "demo_nodes_cpp::OneOffTimerNode"
+  FROM timers_library
+)
+
+rclcpp_components_add_executable(one_off_timer
```

Comment:
Well, `add_executable` takes sources while `rclcpp_components_add_executable` would take components in already built libraries. That's assuming no form of install takes place, which I think it shouldn't. But it may still not be the best naming. I just mimic'd CMake as I had no better ideas.

Reply:
It also creates a new target out of a source file the user doesn't have control over... There's already magic there.

Reply:
Well, that _doesn't have to_ be the case :)

I don't want to make it overly complicated, with user predefined templates and stuff. I just saw the opportunity to confine boilerplate code to a single place while still allowing the user to tinker with the target.